src/scipp/core/dimensions.py

```diff
@@ -1,12 +1,57 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
 # @author Simon Heybrock
-from typing import Dict
+from typing import Dict, Optional
 
 from .._scipp.core import Variable, DataArray, Dataset, CoordError
+from ..typing import VariableLikeType
 from .dataset import merge
 
 
+def _combine_dims(dims_dict: Optional[Dict[str, str]],
+                  names: Dict[str, str]) -> Dict[str, str]:
+    dims_dict = {} if dims_dict is None else dims_dict
+    if set(dims_dict).intersection(names):
+        raise ValueError(
+            'The names passed in the dict and as keyword arguments must be distinct.'
+            f'Got dims_dict={dims_dict} and kwargs={names}')
+    return {**dims_dict, **names}
+
+
+def _rename_dims(self: VariableLikeType,
+                 dims_dict: Optional[Dict[str, str]] = None,
+                 /,
+                 **names: str) -> VariableLikeType:
+    """Rename dimensions.
+
+    The renaming can be defined:
+
+    - using a dict mapping the old to new names, e.g.
+     ``rename_dims({'x': 'a', 'y': 'b'})``
+    - using keyword arguments, e.g. ``rename_dims(x='a', y='b')``
+
+    In both cases, x is renamed to a and y to b.
+
+    Dimensions not specified in either input are unchanged.
+
+    This function only renames dimensions.
+    See the ``rename`` method to also rename coordinates and attributes.
```

Comment:
Can we not link to methods?

Reply:
We can link to all 3 at once. But since this function is used for all 3 classes, it is not so easy to customise the docstring.
See also the type hints.